### ISSUE TITLE
不能复现论文中的结果

### ISSUE THREAD

**Issue by Dawnstar2018** [Treedy2020/TransNet#2]
您好，我下载了相关的代码，遇到下面几个问题，如果有时间期待您的解答，谢谢！
1. 使用COST2100中的测试集，load 您提供的pretraind的模型，无法复现论文中的结果（结果NMSE=22.07db）；
2. 使用COST2100中的训练集进行模型训练，与论文中的参数配置一致的情况下，训练400个epoch，无法复现出论文的结果（论文中压缩率1/4时NMSE为-29db左右，我训练的结果为-26db左右）。

**Comment by guohuayan** [comment]
> 您好，我下载了相关的代码，遇到下面几个问题，如果有时间期待您的解答，谢谢！
> 
> 1. 使用COST2100中的测试集，load 您提供的pretraind的模型，无法复现论文中的结果（结果NMSE=22.07db）；
> 2. 使用COST2100中的训练集进行模型训练，与论文中的参数配置一致的情况下，训练400个epoch，无法复现出论文的结果（论文中压缩率1/4时NMSE为-29db左右，我训练的结果为-26db左右）。

您好，请问在什么地方下载pretraind的模型呢，没有找到。
可以发给我邮箱guohuayan@pku.edu.cn么，非常感谢！
我也在尝试复现这个文章，欢迎交流仿真结果

**Comment by guohuayan** [comment]
> 您好，我下载了相关的代码，遇到下面几个问题，如果有时间期待您的解答，谢谢！
> 
> 1. 使用COST2100中的测试集，load 您提供的pretraind的模型，无法复现论文中的结果（结果NMSE=22.07db）；
> 2. 使用COST2100中的训练集进行模型训练，与论文中的参数配置一致的情况下，训练400个epoch，无法复现出论文的结果（论文中压缩率1/4时NMSE为-29db左右，我训练的结果为-26db左右）。

哦，我找到了，他存在google硬盘里。。不好意思打搅啦。
我刚起了个程序，有结果了给你说

**Comment by CastleJin** [comment]
Hello, guys.

I wanted to use the -29dB pretrained model, 

so I downloaded the indoor 1/4 case model from Google Drive.

However, it works weird.

![Image](https://github.com/user-attachments/assets/4f7eb61b-73d0-448b-b863-9cc7699d2fa8)

NMSE shows 22.29db.

---------------------------

After that, I retrained the model for 400 epochs

and this time I got about -26dB also.

This repository does not match reported result in the original paper. Very weird.

**Comment by huhbak** [comment]
I am in a similar situation as CastleJin and am struggling to reproduce the results from the paper.
Despite investing a significant amount of time, I have been unable to achieve the expected results.
I have tried everything I could.
At this point, I am unsure whether I made a mistake or if there is an issue with the code.

**Comment by CastleJin** [comment]
Hi, @huhbak. 

Afterward, I experimented with Transnet using cosine annealing at epochs 1000 and 10000. 

I achieved -29 dB and -32 dB, respectively. 

I hope these results assist with your research.

**Comment by huhbak** [comment]
Hello, @CastleJin 
Thanks for your kind comment.
Is my understanding below correct in terms of NMSE?

results from the paper (expected result)
1) indoor, 1/4, -29.22, 35.72M, 400epochs
2) indoor, 1/4, -32.38, 35.72M, 1000epochs

castleJin's results
1) indoor, 1/4, -29, 35.72M, 1000epochs
2) indoor, 1/4, -32, 35.72M, 10000epochs

If this is true, then the claim made in the paper may not be accurate..

**Comment by CastleJin** [comment]
Yes, cosine annealing and the use of a large number of epochs are key elements in my case.

However, it is important to acknowledge that neural network training is inherently a stochastic process.

To ensure reproducibility, one can examine the randomness in the code, such as by reviewing the training seeds.

Therefore, the results presented in the paper have the potential to be reproduced.

I look forward to hearing from the others.

**Comment by huhbak** [comment]
Dear @CastleJin 

Thank you for your valuable insights. I have one more question.

In my case, I ran the 4_in.pth checkpoint provided by the author, and the NMSE result was 22.1 dB.

I saw your previous comment mentioning that the NMSE was 22.29 dB. Have you resolved this issue?

I find it difficult to understand why the result consistently remains at a positive 22.1 dB.

**Comment by CastleJin** [comment]
It remains mystery why the pretrained network reports the 22dB result. haha...

**Comment by CastleJin** [comment]
Hello, guys.

I experimented many times. I got -31dB with 1000 epochs.

![Image](https://github.com/user-attachments/assets/62e47c87-09e4-45fc-b718-df3f3c63ea59)

**Comment by huhbak** [comment]
Dear @CastleJin 
Just for your information. Below are results from my experiment.

[Training 1]
  --scenario "in" \
  --batch-size 200 \
  --workers 0 \
  --gpu 0 \
  --epochs 400 \
  --cr 4 \
  --d_model 32 \
  --scheduler const

=! Final test loss: 1.183e-06
         test rho: 9.960e-01
         test NMSE: -2.621e+01

[Training 2]
  --scenario "in" \
  --batch-size 200 \
  --workers 3 \
  --gpu 0 \
  --epochs 1000 \
  --cr 4 \
  --d_model 32 \
  --scheduler cosine

=! Final test loss: 4.606e-07
         test rho: 9.967e-01
         test NMSE: -3.057e+01
